From 97358ea269b64b849a17d7cc361a784c74f57941 Mon Sep 17 00:00:00 2001
Message-ID: <97358ea269b64b849a17d7cc361a784c74f57941.1711915492.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1711915492.git.sharlatanus@gmail.com>
References: <cover.1711915492.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 31 Mar 2024 11:50:07 +0100
Subject: [PATCH 02/12] gnu: Add go-github-com-go-logr-logr.

* gnu/packages/golang-xyz.scm (go-github-com-go-logr-logr): New
variable.

Change-Id: I0fe5e817fa653bcdbe2fea03fe5d608ea270fa12
---
 gnu/packages/golang-xyz.scm | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index 7fe750f59e..1bf296bda0 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -991,6 +991,42 @@ (define-public go-github-com-gabriel-vasile-mimetype
 @end itemize")
     (license license:expat)))
 
+(define-public go-github-com-go-logr-logr
+  (package
+    (name "go-github-com-go-logr-logr")
+    (version "1.4.1")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/go-logr/logr")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "0x0q9jkk2p5pz4lii1qs8ifnsib4ib5s8pigmjwdmagl976g8nhm"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:go go-1.18
+      #:import-path "github.com/go-logr/logr"
+      #:phases
+      #~(modify-phases %standard-phases
+          (add-after 'unpack 'remove-examples-and-benchmarks
+            (lambda* (#:key import-path #:allow-other-keys)
+              (delete-file-recursively
+               (string-append "src/" import-path "/examples"))
+              (delete-file-recursively
+               (string-append "src/" import-path "/funcr/example"))
+              (delete-file-recursively
+               (string-append "src/" import-path "/benchmark")))))))
+    (home-page "https://github.com/go-logr/logr")
+    (synopsis "Minimal logging API for Go")
+    (description
+     "Package @code{logr} defines a general-purpose logging API and abstract
+interfaces to back that API.  Packages in the Go ecosystem can depend on it,
+while callers can implement logging with whatever backend is appropriate.")
+    (license license:asl2.0)))
+
 (define-public go-github-com-hashicorp-errwrap
   (package
     (name "go-github-com-hashicorp-errwrap")
-- 
2.41.0

